arangod/Aql/BindParameters.cpp

```diff
@@ -65,28 +66,42 @@ uint64_t BindParameters::hash() const {
   return _builder->slice().hash();
 }
 
-/// @brief mark a bind parameter as "used", and return its value.
+/// @brief return a bind parameter value by name
 /// will return VPackSlice::noneSlice() if the bind parameter does not exist!
-VPackSlice BindParameters::markUsed(std::string const& name) noexcept {
+std::pair<VPackSlice, AstNode*> BindParameters::get(
+    std::string const& name) const noexcept {
   TRI_ASSERT(_processed);
 
   auto it = _parameters.find(name);
   if (it == _parameters.end()) {
-    return VPackSlice::noneSlice();
+    return std::make_pair(VPackSlice::noneSlice(), nullptr);
   }
 
-  // mark the bind parameter as being used
-  (*it).second.second = true;
+  // return parameter value and AstNode
+  TRI_ASSERT(!(*it).second.first.isNone());
+  return std::make_pair((*it).second.first, (*it).second.second);
```

Comment:
Isn't this just the same as
```suggestion
  return it->second;
```

Reply:
Absolutely... :facepalm: